kirby/api/ext.py

```diff
@@ -198,32 +198,24 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self._session.close()
 
-    @tenacity.retry(**webserver_retry_args)
-    def post(self, endpoint, data, params=None):
-        if not params:
-            params = {}
-        result = self._session.post(
-            urljoin(self.web_endpoint_base, endpoint), data=data, params=params
-        )
-        if result.status_code == 200:
-            return result.json()
-        raise WebClientError(
-            f"POST error on {result.url}. "
-            f"Status code : {result.status_code}. "
-            f"Response : {result.text}"
-        )
+    def _request_decorator(self, method):
+        @tenacity.retry(**webserver_retry_args)
```

Comment:
you do not **need** to use tenacity.retry as an `@decorator`, you could juste 

```python
def request(self, endpoint, method):
    return tenacity.retry(**webserver_retry_args)(method(...)
```

Reply:
It is now written as you proposed